### DISCUSSION TITLE
Composer used by Deployer 7 internally uses the wrong PHP version. What am I doing wrong?

### DISCUSSION THREAD

**Discussion by timmekesegers** [deployphp/deployer#3711] [Help needed] [answered]
Hi.

I am currently looking into upgrading our deployment flow to Deployer 7. As all of our current projects are still being deployed with v6, I'm running them side by side at the moment, aliasing `dep7` to the correct executable:

```
➜  which dep
/usr/local/bin/dep

➜  which dep7
dep7: aliased to /usr/local/bin/dep7

➜  dep -v
Deployer 6.9.0
[...]

➜  dep7 -v
Composer detected issues in your platform:

Your Composer dependencies require a PHP version ">= 7.3.0". You are running 7.2.34.
```

However, I can't interact with v7 at the moment because it complains about my PHP version. While my OS X laptop shipped with 7.2 back in the day, that has long since been updated. All of this is configured in my local setup:

```
➜  ~ php -v
PHP 8.1.1 (cli) (built: Dec 21 2021 17:08:32) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.1, Copyright (c) Zend Technologies

➜  ~ which php
php: aliased to php81

➜  ~ which php81
php81: aliased to /Applications/MAMP/bin/php/php8.1.1/bin/php -c "/Library/Application Support/appsolute/MAMP PRO/conf/php8.1.1.ini"
```

As such, I'm at a bit of a loss. I tried bash profiling this to see if that would help, but alas:

```
➜  /bin/bash -l -c "dep7"
Composer detected issues in your platform:

Your Composer dependencies require a PHP version ">= 7.3.0". You are running 7.2.34.
```

Is there a way to tell Deployer which PHP version to use? I'm familiar with the ability to define the PHP and Composer executables etc when deploying to a server, but determining which PHP version Deployer itself uses eludes me.

Any and all help appreciated!

**Comment by timmekesegers** [comment, marked as answer]
I found it. Deployer uses the default PHP version, which is located here:

```
/usr/local/bin/php
```

Turns out that still pointed to the very old 7.2:

```
➜  ~ /usr/local/bin/php -v
PHP 7.2.34 (cli) (built: Sep 30 2021 12:59:02) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies

➜  ~ readlink /usr/local/bin/php
/Applications/MAMP/bin/php/php7.2.34/bin/php 
```

So the solution to my problem was to simply update that symlink to 8.1:

```
➜  ~ rm /usr/local/bin/php && ln -s /Applications/MAMP/bin/php/php8.1.1/bin/php /usr/local/bin/php

➜  ~ dep7
Deployer 7.3.0
[...]
```

Case closed :) But figured I'd comment it here in case anyone else bumps into this problem.

**Reply by fd6130** [reply]
Is there a way to set deployer to run at which php version during deploy?

**Reply by antonmedv** [reply]
```php
run('which php');
```
